### DISCUSSION TITLE
Lesson 4: brownie init

### DISCUSSION THREAD

**Discussion by CrispyButter** [smartcontractkit/full-blockchain-solidity-course-py#508] [Q&A]
I installed Brownie, ran brownie --version, and it retuned v1.16.4,

But I get a FileExistsError when trying to run brownie init. Any advice on fixing the below:
(venv) Trell@Kentrells-MacBook-Pro brownie % brownie --version
Brownie v1.16.4 - Python development framework for Ethereum

(venv) Trell@Kentrells-MacBook-Pro brownie % brownie
Brownie v1.16.4 - Python development framework for Ethereum

Usage:  brownie <command> [<args>...] [options <args>]

Commands:
  init               Initialize a new brownie project
  bake               Initialize from a brownie-mix template
  pm                 Install and manage external packages
  compile            Compile the contract source files
  console            Load the console
  test               Run test cases in the tests/ folder
  run                Run a script in the scripts/ folder
  accounts           Manage local accounts
  networks           Manage network settings
  gui                Load the GUI to view opcodes and test coverage
  analyze            Find security vulnerabilities using the MythX API

Options:
  --help -h          Display this message
  --version          Show version and exit

Type 'brownie <command> --help' for specific options and more information about
each command.
(venv) Trell@Kentrells-MacBook-Pro brownie % brownie init
Brownie v1.16.4 - Python development framework for Ethereum

  File "brownie/_cli/__main__.py", line 64, in main
    importlib.import_module(f"brownie._cli.{cmd}").main()
  File "brownie/_cli/init.py", line 31, in main
    path = project.new(args["<path>"] or ".", args["--force"], args["--force"])
  File "brownie/project/main.py", line 584, in new
    raise FileExistsError(f"Directory is not empty: {project_path}")
FileExistsError: Directory is not empty: /Users/Trell/demos/web3_py_simple_storage/demos/brownie_simple_storage/brownie
(venv) Trell@Kentrells-MacBook-Pro brownie %

**Comment by cromewar** [comment]
hello @CrispyButter brownie requires a empty folder to create a new project, so please create a new folder and create the project there.
Best regards, Cromewar.

**Reply by CrispyButter** [reply]
How exactly do I create a brand new file? I thought mkdir brownie_simple_storage was a new folder? Do you mean create a new project all together? If, so do I have to activate a new virtual environment and install py-solcx and all the other things as well?

**Reply by CrispyButter** [reply]
Sorry, brand new folder

**Reply by cromewar** [reply]
I mean your folder just must to be empty, that's all. And yes creating a folder with mkdir should work. As you are using python envs, do not use brownie init on the folder you have the env, I would recommend to have two folders:

- One for the env (virtualenv envName)
- other for the project (mkdir dev)

so move to the dev folder (cd dev) and start the project there with `brownie init`.

Give it a try.

**Comment by CrispyButter** [comment]
I think it might be because I pip install cython and then pip install cytoolz after receiving the error below. and after installing those I THEN did pip install eth-brownie which worked. but then I ran into the original problem when trying to do brownie init. Any help or if I can set aside time for a video conference that would be awesome. I really want to move forward on the course


ERROR: Command errored out with exit status 1:
     command: /Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/h5/p88j1qg142j_z6q2pwtkk2n80000gn/T/pip-install-02z_xquo/cytoolz_a51cf8b439144dddb94cbded0d7a8cf6/setup.py'"'"'; __file__='"'"'/private/var/folders/h5/p88j1qg142j_z6q2pwtkk2n80000gn/T/pip-install-02z_xquo/cytoolz_a51cf8b439144dddb94cbded0d7a8cf6/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/h5/p88j1qg142j_z6q2pwtkk2n80000gn/T/pip-record-w37qavbl/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.10/include/python3.10/cytoolz
         cwd: /private/var/folders/h5/p88j1qg142j_z6q2pwtkk2n80000gn/T/pip-install-02z_xquo/cytoolz_a51cf8b439144dddb94cbded0d7a8cf6/
    Complete output (75 lines):
    ALERT: Cython not installed.  Building without Cython.
    running install
    running build
    running build_py
    creating build
    creating build/lib.macosx-10.9-universal2-3.10
    creating build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/compatibility.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/utils_test.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/_version.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/__init__.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/_signatures.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    creating build/lib.macosx-10.9-universal2-3.10/cytoolz/curried
    copying cytoolz/curried/operator.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/curried
    copying cytoolz/curried/__init__.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/curried
    copying cytoolz/curried/exceptions.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/curried
    copying cytoolz/itertoolz.pyx -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/dicttoolz.pyx -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/functoolz.pyx -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/recipes.pyx -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/utils.pyx -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/utils.pxd -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/__init__.pxd -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/recipes.pxd -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/functoolz.pxd -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/dicttoolz.pxd -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/cpython.pxd -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    copying cytoolz/itertoolz.pxd -> build/lib.macosx-10.9-universal2-3.10/cytoolz
    creating build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_none_safe.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_utils.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_curried.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_compatibility.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_embedded_sigs.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_functoolz.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_inspect_args.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_doctests.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_tlz.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_signatures.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/dev_skip_test.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_recipes.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_docstrings.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_dev_skip_test.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_dicttoolz.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_serialization.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_curried_toolzlike.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    copying cytoolz/tests/test_itertoolz.py -> build/lib.macosx-10.9-universal2-3.10/cytoolz/tests
    running build_ext
    building 'cytoolz.dicttoolz' extension
    creating build/temp.macosx-10.9-universal2-3.10
    creating build/temp.macosx-10.9-universal2-3.10/cytoolz
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g -I/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10 -c cytoolz/dicttoolz.c -o build/temp.macosx-10.9-universal2-3.10/cytoolz/dicttoolz.o
    clang -bundle -undefined dynamic_lookup -arch arm64 -arch x86_64 -g build/temp.macosx-10.9-universal2-3.10/cytoolz/dicttoolz.o -o build/lib.macosx-10.9-universal2-3.10/cytoolz/dicttoolz.cpython-310-darwin.so
    building 'cytoolz.functoolz' extension
    clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -arch arm64 -arch x86_64 -g -I/Library/Frameworks/Python.framework/Versions/3.10/include/python3.10 -c cytoolz/functoolz.c -o build/temp.macosx-10.9-universal2-3.10/cytoolz/functoolz.o
    cytoolz/functoolz.c:23087:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                      ^
    cytoolz/functoolz.c:23087:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    cytoolz/functoolz.c:23092:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                      ^
    cytoolz/functoolz.c:23092:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, value == Py_None ? NULL : value);
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    cytoolz/functoolz.c:23176:19: error: implicit declaration of function '_PyGen_Send' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
                ret = _PyGen_Send((PyGenObject*)yf, NULL);
                      ^
    cytoolz/functoolz.c:23176:17: warning: incompatible integer to pointer conversion assigning to 'PyObject *' (aka 'struct _object *') from 'int' [-Wint-conversion]
                ret = _PyGen_Send((PyGenObject*)yf, NULL);
                    ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    3 warnings and 3 errors generated.
    error: command '/usr/bin/clang' failed with exit code 1
    ----------------------------------------
  Rolling back uninstall of cytoolz
  Moving to /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cytoolz-0.11.2-py3.10.egg-info
   from /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/~ytoolz-0.11.2-py3.10.egg-info
  Moving to /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/cytoolz/
   from /Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/~ytoolz
ERROR: Command errored out with exit status 1: /Library/Frameworks/Python.framework/Versions/3.10/bin/python3.10 -u -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/h5/p88j1qg142j_z6q2pwtkk2n80000gn/T/pip-install-02z_xquo/cytoolz_a51cf8b439144dddb94cbded0d7a8cf6/setup.py'"'"'; __file__='"'"'/private/var/folders/h5/p88j1qg142j_z6q2pwtkk2n80000gn/T/pip-install-02z_xquo/cytoolz_a51cf8b439144dddb94cbded0d7a8cf6/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /private/var/folders/h5/p88j1qg142j_z6q2pwtkk2n80000gn/T/pip-record-w37qavbl/install-record.txt --single-version-externally-managed --compile --install-headers /Library/Frameworks/Python.framework/Versions/3.10/include/python3.10/cytoolz Check the logs for full command output.

**Reply by CrispyButter** [reply]
I figured it out somehow!

**Reply by cromewar** [reply]
Hello @CrispyButter could you share how you solved the problem in a comment bellow? This would help other users who might face the same problem.
